command/alloc_exec.go

```diff
@@ -0,0 +1,338 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker/pkg/term"
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/api/contexts"
+	"github.com/posener/complete"
+)
+
+type AllocExecCommand struct {
+	Meta
+
+	Stdin  io.Reader
+	Stdout io.WriteCloser
+	Stderr io.WriteCloser
+}
+
+func (l *AllocExecCommand) Help() string {
+	helpText := `
+Usage: nomad alloc exec [options] <allocation> <command>
+
+  Run command inside the environment of the given allocation and task.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Exec Specific Options:
+
+  -task <task-name>
+    Sets the task to exec command in
+
+  -job <job-id>
+    Use a random allocation from the specified job ID.
+
+  -i, --stdin=true
```

Comment:
We tend to not have short and long flag seperations in hashi tools, and `mitchellh/cli` afaik makes it impossible to differentiate `-` and `--`?

Reply:
Yes - but it has been a goal to support `-i` and `-t` to match folks experience with docker.

Reply:
We might want to just have `-i` and `-t` (or at least unify on `-` for the aliases like the rest of nomad)?

Reply:
done - made it `-i` and `-t` only.